lib/ansible/plugins/connection/paramiko_ssh.py

```diff
@@ -72,6 +72,10 @@
         pass
 
 
+class AnsibleParamikoConnectionFailure(AnsibleConnectionFailure):
```

Comment:
Initial motivation for the more specific exceptions in general was time spent attempting to track down a connection error that could have been raised from multiple places. [AnsibleConnectionFailure('SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh')] 

Generally my preference is to raise the most specific exception I can, especially if there is a reasonable expectation that the exception could be handled differently than it's base exception. But even if it's only to convey additional context its useful.

Granted, at the moment, this module or the app doesn't make any use of this specific exception (aside from the places the exception name is exposed to the user). 

Conceptually, the app could make use of it (for example, a smart connection type could allow a failed paramiko connection to be retried as a ssh connection or vice versa). Long term the connection type specific exceptions will be useful in nested or stacked connection scenarios.